src/classes/PS_ElevateAPI.cls

```diff
@@ -0,0 +1,242 @@
+/*
+    Copyright (c) 2020, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Elevate Payment Services
+* @description Contains common Elevate payment services API methods to send HttpRequest and return response
+*/
+public inherited sharing class PS_ElevateAPI {
```

Comment:
@andrewyu-salesforce , this is a common Elevate Payment Services API class created from extracting GE Payment Services code with a change of adding the Commitment end point and making `sendRequest` generic. The class might change after working on the commitment API request though, will commit changes if so.

Reply:
@andrewyu-salesforce , this version can be used to send the request. The Commitment Cancel request can be sent as:
```
String commitmentId = '<specify the commitment Id>';
PS_ElevateAPI.ResponseWrapper response = new PS_ElevateAPI.RequestBuilder()
    .withMethod(PS_ElevateAPI.HttpRequestMethod.POST)
    .withEndpoint(PS_ElevateAPI.ElevateEndpoint.COMMITMENT_CANCEL)
    .withCommitmentId(commitmentId)
    .sendRequest();
```